File: web/projects/portal/src/app/vsobjects/model/interval-pane-model.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { DynamicValueModel } from "./dynamic-value-model";

export interface IntervalPaneModel {
   level: DynamicValueModel;
   granularity: DynamicValueModel;
   endDayAsToDate: boolean;
   intervalEndDate: DynamicValueModel;
   inclusive: boolean;
   contextLevel: DynamicValueModel;
}

export const enum IntervalLevel {
   DAY = 0x1,
   WEEK = 0x2,
   MONTH = 0x4,
   QUARTER = 0x8,
   YEAR = 0x10,

   TO_DATE = 0x20,
   WEEK_TO_DATE = TO_DATE | WEEK,
   MONTH_TO_DATE = TO_DATE | MONTH,
   QUARTER_TO_DATE = TO_DATE | QUARTER,
   YEAR_TO_DATE = TO_DATE | YEAR,

   SAME_DATE = 0x40,
   SAME_DAY = SAME_DATE | DAY,
   SAME_WEEK = SAME_DATE | WEEK,
   SAME_MONTH = SAME_DATE | MONTH,
   SAME_QUARTER = SAME_DATE | QUARTER
}

File: core/src/main/java/inetsoft/web/composer/model/vs/IntervalPaneModel.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.composer.model.vs;

import inetsoft.uql.viewsheet.internal.DateComparisonInfo;
import inetsoft.uql.viewsheet.internal.DateComparisonInterval;

public class IntervalPaneModel {
   public IntervalPaneModel() {
   }

   public IntervalPaneModel(DateComparisonInterval interval) {
      super();

      if(interval != null) {
         setInclusive(interval.isInclusive());
         setIntervalEndDate(new DynamicValueModel(interval.getIntervalEndDateValue()));
         setLevel(new DynamicValueModel(interval.getLevelValue()));
         setGranularity(new DynamicValueModel(interval.getGranularityValue()));
         setEndDayAsToDate(interval.isEndDayAsToDate());
         setContextLevel(new DynamicValueModel(interval.getContextLevelValue()));
      }
   }

   public DateComparisonInterval toDateComparisonInterval() {
      DateComparisonInterval interval = new DateComparisonInterval();
      interval.setInclusive(isInclusive());

      if(getIntervalEndDate() != null) {
         interval.setIntervalEndDateValue(getIntervalEndDate().convertToValue());
      }

      if(getLevel() != null) {
         interval.setLevelValue(getLevel().convertToValue());
      }

      if(getContextLevel() != null) {
         interval.setContextLevelValue(getContextLevel().convertToValue());
      }

      if(getGranularity() != null) {
         interval.setGranularityValue(getGranularity().convertToValue());
      }

      interval.setEndDayAsToDate(isEndDayAsToDate());

      return interval;
   }

   public DynamicValueModel getLevel() {
      return level;
   }

   public void setLevel(DynamicValueModel level) {
      this.level = level;
   }

   public DynamicValueModel getGranularity() {
      return granularity;
   }

   public void setGranularity(DynamicValueModel granularity) {
      this.granularity = granularity;
   }

   public boolean isEndDayAsToDate() {
      return endDayAsToDate;
   }

   public void setEndDayAsToDate(boolean endDayAsToDate) {
      this.endDayAsToDate = endDayAsToDate;
   }

   public DynamicValueModel getIntervalEndDate() {
      return intervalEndDate;
   }

   public void setIntervalEndDate(DynamicValueModel intervalEndDate) {
      this.intervalEndDate = intervalEndDate;
   }

   public boolean isInclusive() {
      return inclusive;
   }

   public void setInclusive(boolean inclusive) {
      this.inclusive = inclusive;
   }

   public DynamicValueModel getContextLevel() {
      return contextLevel;
   }

   public void setContextLevel(DynamicValueModel contextLevel) {
      this.contextLevel = contextLevel;
   }

   private boolean endDayAsToDate = true;
   private boolean inclusive = true;
   private DynamicValueModel level = new DynamicValueModel(DateComparisonInfo.YEAR,
      DynamicValueModel.VALUE);
   private DynamicValueModel granularity = new DynamicValueModel(DateComparisonInfo.YEAR,
                                                                 DynamicValueModel.VALUE);
   private DynamicValueModel intervalEndDate = new DynamicValueModel(null, DynamicValueModel.VALUE);
   private DynamicValueModel contextLevel = new DynamicValueModel(DateComparisonInfo.YEAR,
      DynamicValueModel.VALUE);

}
